### PR TITLE
Fixes hipRTC after changes to f8

### PR DESCRIPTION
**Pull request by cgmillette**
- Staging some fixes here to reflect some changes coming to f8 headers.
- Fixes some inclusion order
- Fixes some type_traits overrides in hipRTC

### CODE REVIEW

**Comment by cgmillette** [comment]
supersedes #442

File: library/include/rocwmma/internal/float8.hpp [hunk 1]
```diff
@@ -120,21 +120,34 @@ namespace ROCWMMA_TYPE_TRAITS_IMPL_NAMESPACE
...
+// 2. Give ROCWMMA_FP8_FNUZ value of 1
+
+//static_assert((bool)ROCWMMA_ARCH_GFX94X || (bool)ROCWMMA_ARCH_HOST,
+//              "fp8_fnuz types only supported on gfx94X archs");
+
+#define ROCWMMA_FP8_FNUZ (ROCWMMA_ARCH_GFX94X || ROCWMMA_ARCH_HOST)
 #define ROCWMMA_FP8_FNUZ_VISIBILITY ROCWMMA_HOST_DEVICE
 #else
 #define ROCWMMA_FP8_FNUZ 0
 #define ROCWMMA_FP8_FNUZ_VISIBILITY ROCWMMA_HOST
```
Comment (therahulmanocha) [inline library/include/rocwmma/internal/float8.hpp:135]: #if defined(__HIPCC_RTC__)
    #define ROCWMMA_FP8_FNUZ_VISIBILITY ROCWMMA_DEVICE
#else
    #define ROCWMMA_FP8_FNUZ_VISIBILITY ROCWMMA_HOST
#endif

Comment (manocharahul) [inline library/include/rocwmma/internal/float8.hpp:135]: #if defined(__HIPCC_RTC__)
    #define ROCWMMA_FP8_FNUZ_VISIBILITY ROCWMMA_DEVICE
#else
    #define ROCWMMA_FP8_FNUZ_VISIBILITY ROCWMMA_HOST
#endif

File: library/include/rocwmma/internal/float8.hpp [hunk 2]
```diff
@@ -120,21 +120,34 @@ namespace ROCWMMA_TYPE_TRAITS_IMPL_NAMESPACE
...
+
+//static_assert((bool)ROCWMMA_ARCH_GFX12 || (bool)ROCWMMA_ARCH_HOST,
+//              "fp8_fnuz types only supported on gfx12 archs");
+
+#define ROCWMMA_FP8 (ROCWMMA_ARCH_GFX12 || ROCWMMA_ARCH_HOST)
 #define ROCWMMA_FP8_VISIBILITY ROCWMMA_HOST_DEVICE
 #else
+
 #define ROCWMMA_FP8 0
 #define ROCWMMA_FP8_VISIBILITY ROCWMMA_HOST
```
Comment (manocharahul) [inline library/include/rocwmma/internal/float8.hpp:152]: #if defined(__HIPCC_RTC__)
    #define ROCWMMA_FP8_VISIBILITY ROCWMMA_DEVICE
#else
    #define ROCWMMA_FP8_VISIBILITY ROCWMMA_HOST
#endif

**Review by manocharahul — APPROVED** [review]

**Review by dlangbe — APPROVED** [review]

**Pull request merged**